code/modules/mob/living/carbon/human/species/station.dm

```diff
@@ -381,8 +387,154 @@
+		if(!istype(E) || istype(E, /obj/item/organ/external/stump))
+			var/list/limblist = species.has_limbs[l]
+			var/obj/item/organ/external/limb = limblist["path"]
+			var/parent_organ = initial(limb.parent_organ)
+			var/obj/item/organ/external/parentLimb = organs_by_name[parent_organ]
+			if(!istype(parentLimb) || parentLimb.is_stump())
+				continue
+			missing_limbs[initial(limb.name)] = l
+
+	if(!missing_limbs.len)
+		src << "<span class='warning'>You're not missing any limbs!</span>"
+		return
+
+	var/limb_select = input(src, "Choose a limb to regrow", "Limb Regrowth") as null|anything in missing_limbs
+	var/chosen_limb = missing_limbs[limb_select]
+
+	visible_message("<span class='notice'>[src] begins to hold still and concentrate on their missing [limb_select]...</span>", "<span class='notice'>You begin to focus on regrowing your missing [limb_select]... (This will take [round(regrowthdelay/10)] seconds)</span>")
+	if(do_after(src, regrowthdelay, needhand=0))
+		if(stat || paralysis || stunned)
+			src << "<span class='warning'>You cannot regenerate missing limbs in your current state.</span>"
+			return
+
+		if(nutrition < minhunger)
+			src << "<span class='warning'>You're too hungry to regenerate a limb!</span>"
+			return
+
+		var/obj/item/organ/external/O = organs_by_name[chosen_limb]
+
+		var/stored_brute = 0
+		var/stored_burn = 0
+		if(istype(O))
+			if(!O.is_stump())
+				src << "<span class='warning'>Your limb has already been replaced in some way!</span>"
+				return
+			else
+				src << "<span class='warning'>You distribute the damaged tissue around your body, out of the way of your new pseudopod!</span>"
+				var/obj/item/organ/external/doomedStump = O
+				stored_brute = doomedStump.brute_dam
+				stored_burn = doomedStump.burn_dam
+				qdel(O)
+
+		var/limb_list = species.has_limbs[chosen_limb]
+		var/obj/item/organ/external/limb_path = limb_list["path"]
+		// Parent check
+		var/obj/item/organ/external/potential_parent = organs_by_name[initial(limb_path.parent_organ)]
+		if(!istype(potential_parent) || potential_parent.is_stump())
+			src << "<span class='danger'>You've lost the organ that you've been growing your new part on!</span>"
+			return // No rayman for you
+		// Grah this line will leave a "not used" warning, in spite of the fact that the new() proc WILL do the thing.
+		// Bothersome.
+		var/obj/item/organ/external/new_limb = new limb_path(src)
+		adjustBruteLoss(stored_brute)
+		adjustFireLoss(stored_burn)
+		update_body()
+		updatehealth()
+		UpdateDamageIcon()
+		nutrition -= hungercost
+		visible_message("<span class='notice'>[src] finishes regrowing their missing [limb_select]!</span>", "<span class='notice'>You finish regrowing your [limb_select]</span>")
+	else
+		src << "<span class='warning'>You need to hold still in order to regrow a limb!</span>"
+	return
+
+/datum/species/slime/handle_post_spawn(var/mob/living/carbon/human/H)
+	..()
+	H.verbs += /mob/living/carbon/human/proc/regrow_limbs
```

Comment:
I might want to polish this bit to make sure everything's working as I mean it to.